src/model.rs

```diff
@@ -1446,6 +1446,78 @@ impl Model {
         self.collections
     }
 }
+#[cfg(feature = "mutable-model")]
+impl Model {
+    /// Add a Calendar inside the model and link it automatically to the corresponding Vehicle Journeys
+    pub fn add_calendar(
+        &mut self,
+        calendar: Calendar,
+        vehicle_journeys: IdxSet<VehicleJourney>,
+    ) -> Result<()> {
```

Comment:
Maybe this can be split into 2 functions. One that add objects into the `Model` (and might create orphan objects) and another one specifically to modify the relations.

Reply:
The function has been split into 2 parts.

```rust
fn add_calendar(&mut self, calendar: Calendar) -> Result<Idx<Calendar>>;
fn connect_calendar_to_vehicle_journeys(
  &mut self,
  calendar_idx: Idx<Calendar>,
  vehicle_journey_idxs: IntoIterator<Item = Idx<VehicleJourney>>,
) -> Result<()>;